### DISCUSSION TITLE
Why "Unable to debug this expression?"

### DISCUSSION THREAD

**Discussion by bjulius** [TabularEditor/TabularEditor3#385] [🙋♀️TE3 Q&A]
![Snag_7f55d6](https://user-images.githubusercontent.com/77697750/151896087-9042dada-0367-44d2-8e4f-eb3180c673eb.png)

I'm working on learning how to use the awesome new DAX Debugger in TE3.  Running most current v. 3.2.1, and not understanding why I'm getting the "Unable to debug this expression" message?  Any insight would be greatly appreciated.

Thanks!

- Brian

**Comment by otykier** [comment]
Hi Brian,
Unfortunately, we do not (yet) support debugging of arbitrary DAX queries. We have focused our efforts on debugging of measures as well as queries commonly generated by Power BI Desktop: https://docs.tabulareditor.com/te3/features/dax-debugger.html#limitations-and-known-issues

We do have plans to support more types of queries over time, but there will always be some limitations to this, as queries do not necessarily return data in any particular order, and a single row of a query result might not hold the information needed to determine filter/row context for the selected cell.

**Reply by bjulius** [reply]
Daniel,

Thanks very much for the prompt and detailed reply.  I get it now.

And this isn't the first time that my DAX has been referred to as "arbitrary".  😆 

- Brian
